docs: Update Showcase Dynamic plugins documentation to remove the part related to the old backend system (#1105)
janus-idp/backstage-showcase: showcase-docs/dynamic-plugins.md

### Backend plugins

To be compatible with the showcase dynamic plugin support, and used as dynamic plugins, existing plugins must be based on, or compatible with, the new backend system, as well as rebuilt with a dedicated CLI command.

#### Add the Dynamic Plugin script

There are some changes to be made to the plugin code, in order to make it compatible with the dynamic plugin support:

1. The plugin must:

- include the `@janus-idp/cli` dependency, which provides a new, required, `export-dynamic-plugin` command.
- include the `export-dynamic` script entry,

These recommended changes to the `package.json` are summarized below:

```json
  ...
  "scripts": {
    ...

>     "export-dynamic": "janus-cli package export-dynamic-plugin ----embed-as-dependencies"

Author: done, thanks!

### Backend plugins

> To be compatible with the showcase dynamic plugin support, and used as dynamic plugins, existing plugins must be completed code-wise, as well as rebuilt with a dedicated CLI command.

Reviewer: ```suggestion
To be compatible with the showcase dynamic plugin support, and used as dynamic plugins, existing plugins must be based on, or compatible with, the new backend system, as well as rebuilt with a dedicated CLI command.
```

  ...
  "dependencies": {
    ...

>     "@backstage/backend-dynamic-feature-service": "0.2.3",

Reviewer: In fact that should not even be necessary in most cases, when using a plugin made for the new backend system.

Reviewer: So line https://github.com/janus-idp/backstage-showcase/pull/1105/files#diff-8d2e5712d5ef532a983602f4c771597e41d8c082d7801f59f6003fe9f66e2a89L28 can be removed.

Line https://github.com/janus-idp/backstage-showcase/pull/1105/files#diff-8d2e5712d5ef532a983602f4c771597e41d8c082d7801f59f6003fe9f66e2a89R28 should also be removed

Reviewer: In line 38 you should use:

```
          "export-dynamic": "janus-cli package export-dynamic-plugin ----embed-as-dependencies"
```

    ...
  }
  ...
  "devDependencies": {

>     "@janus-idp/cli": "^1.7.5"

Reviewer: ```suggestion
    "@janus-idp/cli": "^1.7.7"
```

  },
  ...

>   "files": [

Reviewer: This required change can be removed now

  ],
```

> #### Creating entry points

Reviewer: ```suggestion
#### Required entry points
```

> It is required to create entry points (using `createBackendPlugin` or `createBackendModule`) to enable a backend plugin's dynamic functionality.

Reviewer: ```suggestion
The new backend system standard entrypoint (created using  `createBackendPlugin()` or `createBackendModule()`) should be exported as the default export of either the main package or of an `alpha` package (if the new backend support is still provided as `apha` APIs).

So this doesn't add any additional requirement on top of the standard plugin development guidelines of the new backend system.
```
